src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -379,6 +386,42 @@ private void InitializeTelemetry(TelemetryActivity telemetry, int httpSourcesCou
             }
 
             telemetry.TelemetryEvent[AuditEnabled] = auditEnabled ? "enabled" : "disabled";
+
+            PopulatePruningEnabledTelemetry(_request.Project, telemetry.TelemetryEvent);
+        }
+
+        internal static void PopulatePruningEnabledTelemetry(PackageSpec project, TelemetryEvent telemetryEvent)
+        {
+            int pruningEnabledCount = 0;
+            int pruningDisabledCount = 0;
+            int pruningNotApplicableCount = 0;
+
+            foreach (var framework in project.TargetFrameworks.NoAllocEnumerate())
+            {
+                bool isPruningEnabled = framework.PackagesToPrune.Count > 0;
```

Comment:
Should this be checking an enabled flag, like `RestoreEnablePackagePruning`?

Reply:
We don't read RestoreEnablePackagePruning, it's only a property. 
In the PackageSpec, we just have a list of prune refs and if they're there we use them, if not, we don't. 
If RestoreEnablePackagePruning is set to false, the PackagesToPrune will be empty.

Reply:
An example of that implementation: https://github.com/NuGet/NuGet.Client/blob/060054eaf7f0ff1d7099bf63b410924709f5c0f1/src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs#L144-L149

Reply:
Ok, as long as you're not needing to know the enabled but nothing to prune case, this works.